pyccel/codegen/printing/ccode.py

```diff
@@ -668,6 +698,11 @@ def _print_FunctionDef(self, expr):
         decs += [Declare(i.dtype, i) for i in self._additional_declare]
         decs  = '\n'.join(self._print(i) for i in decs)
         self._additional_declare.clear()
+
+        if len(expr.results) > 1:
+            self._additional_args = [a.clone(name = self._parser.get_new_name('tmp_ret'), is_pointer =True) for a in expr.results]
```

Comment:
Why do the variables need renaming?

Reply:
it's not renaming I'm creating a list of new temp variables

Reply:
Ok. Can you post a code snippet so I can see what this generates? I think results are already temporary variables so I'm worried you are going to get something like:
```
def f():
    a = 1
    b = 3.0
    return a, b
```
```
int f(int* tmp_ret_1, float* tmp_ret_2)
{
    int a = 1;
    int b = 3.0;
    int Out_1 = a;
    float Out_2 = b;
    (*tmp_ret_1) = Out_1;
    (*tmp_ret_2) = Out_2;
}
```

Reply:
i didn't really understand  but the generated code  is like the one that you write 
```int f(int* tmp_ret_1, float* tmp_ret_2)
{
    int a = 1;
    int b = 3.0;
    int Out_1 = a;
    float Out_2 = b;
    (*tmp_ret_1) = Out_1;
    (*tmp_ret_2) = Out_2;
}
```

Reply:
The variables `Out_X` are created because in fortran all returns need to have the same name.

In c or python we can do:
```
def f(b):
    a = 1
    c = 2
    if b:
        return a
    else:
        return c
```
but in fortran this is not possible. So these `Out_X` variables were created so that the code becomes:
```
def f(b):
    a = 1
    c = 2
    if b:
        Out_1 = a
        return Out_1
    else:
        Out_1 = c
        return Out_1
```
These `Out_X` variables are not used anywhere else They are just temporary variables for the output.
Your `tmp_ret_X` variables do the same job. So you will be doing the same thing twice. The code above would generate the following if you used similar logic with 1 return:
```
def f(b):
    a = 1
    c = 2
    if b:
        Out_1 = a
        (*tmp_ret_1) = Out_1
        return tmp_ret_1
    else:
        Out_1 = c
        (*tmp_ret_1) = Out_1
        return tmp_ret_1
```
It shouldn't be necessary to create a new temporary variable. You should be able to use the existing `Out_X` variables. So the example from before should create:
```
int f(int* Out_1, float* Out_2)
{
    int a = 1;
    int b = 3.0;
    (*Out_1) = a;
    (*Out_2) = b;
}
```
which is cleaner

Reply:
The difficulty is how to implement this ^^'. Ideally there should be some way of knowing that the `Out_X` variables are part of a multiple return, then you can just add them to `stored_in_c_pointer`

Reply:
in this case, is it safe to use Variable class setter to change the is_pointer property?

Reply:
I think in this case it is maybe ok